### PR TITLE
fix: build size limits

### PR DESCRIPTION
**Pull request by loks0n**
- introduces a new environment variable `_APP_FUNCTIONS_BUILD_SIZE_LIMIT`
- adds size limit check in Builds.php and throws an exception if the built deployment size exceeds the limit

### CODE REVIEW

File: app/config/variables.php [hunk 1]
```diff
@@ -718,6 +718,15 @@
                 'question' => '',
                 'filter' => ''
             ],
+            [
+                'name' => '_APP_FUNCTIONS_SIZE_LIMIT',
```
Comment (christyjacob4) [inline app/config/variables.php:722]: we already have the same variable above?

Comment (loks0n) [reply app/config/variables.php]: whoops, fixed

File: app/config/variables.php [hunk 2]
```diff
@@ -711,13 +711,22 @@
...
                 'default' => '30000000',
                 'required' => false,
                 'question' => '',
                 'filter' => ''
             ],
+            [
+                'name' => '_APP_FUNCTIONS_BUILD_SIZE_LIMIT',
+                'description' => 'The maximum size of a built deployment in bytes. The default value is 2GB.',
+                'introduction' => '1.6.0',
+                'default' => '2000000000',
```
Comment (stnguyen90) [inline app/config/variables.php:725]: If this is higher than builds.size:

https://github.com/appwrite/appwrite/blob/4b1bf59091b747c5855c355b6f48a5346b1cbe0c/app/config/collections.php#L3559-L3569

there will still be an error, right?

I think we need to update the collection to make the size bigger than 8 or unsigned.

Comment (byawitz) [reply app/config/variables.php:725]: The value of `builds.size` is up to [4294967295](https://github.com/utopia-php/database/blob/c712d1f6c8ec37886a7a1ad4d60a8cd75dec00aa/src/Database/Adapter/SQL.php#L185), so we're good on that part.

Comment (stnguyen90) [reply app/config/variables.php:725]: What if someone tries to increase this env var to bigger than that?

Comment (loks0n) [reply app/config/variables.php:725]: I've added this as a note to the description of the variable. Users changing this value can refer to this, I don't think there's any more value in doing validation here.

**Review by byawitz — APPROVED** [review]

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**